api/jobs.go

```diff
@@ -119,12 +120,51 @@ func (api *JobStoreAPI) GetJobsHandler(w http.ResponseWriter, req *http.Request)
 		http.Error(w, serverErrorMessage, http.StatusInternalServerError)
 		return
 	}
-
 }
 
 // unlockJob unlocks the provided job lockID and logs any error with WARN state
 func (api *JobStoreAPI) unlockJob(ctx context.Context, lockID string) {
+	log.Event(ctx, "Entering unlockJob function, which unlocks the provided job lockID.", log.INFO)
 	if err := api.jobStore.UnlockJob(lockID); err != nil {
 		log.Event(ctx, "error unlocking lockID for a job resource", log.WARN, log.Data{"lockID": lockID})
 	}
 }
+
+// PutNumTasksHandler returns a function that updates the number_of_tasks in an existing Job resource, which is associated with the id passed in.
+func (api *JobStoreAPI) PutNumTasksHandler(ctx context.Context) http.HandlerFunc {
+	log.Event(ctx, "Creating handler function, which calls PutNumberOfTasks to update the number_of_tasks in the job with the supplied id.", log.INFO)
+	return func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		vars := mux.Vars(req)
+		id := vars["id"]
+		count := vars["count"]
+
+		var numTasks int
+		numTasks, err := strconv.Atoi(count)
+		logData := log.Data{"job_id": id, "num_tasks": numTasks}
+
+		if err != nil {
+			log.Event(ctx, "invalid path parameter - count should be an integer", log.Error(err), logData, log.ERROR)
+			http.Error(w, "invalid path parameter - count should be an integer", http.StatusBadRequest)
+			return
+		}
+
+		lockID, err := api.jobStore.AcquireJobLock(ctx, id)
+		if err != nil {
+			log.Event(ctx, "acquiring lock for job ID failed", log.Error(err), logData, log.ERROR)
+			http.Error(w, serverErrorMessage, http.StatusInternalServerError)
+			return
+		}
+		defer api.unlockJob(ctx, lockID)
+
+		err = api.jobStore.PutNumberOfTasks(req.Context(), id, numTasks)
+		if err != nil {
+			log.Event(ctx, "putting number of tasks failed", log.Error(err), logData, log.ERROR)
+			http.Error(w, "Failed to find job in job store", http.StatusNotFound)
```

Comment:
Is `"Failed to find job in job store"` accurate for this error? Also should is not found status the most appropriate state code for this scenario?

Reply:
Good spot. I think this was laziness on my part. I've updated it now so that the "Failed to find job in job store" message will only occur if that is definitely what has gone wrong. Otherwise it will now give an internal server error if something unexpected occurs. I've updated the unit tests for this too.